Added an Infiltrate function to lua

abcdefg30 opened the pull request.

pchote commented on OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs:
```diff
@@ -0,0 +1,43 @@
+			: base(context, self) { }
+
+		[Desc("Captures the target actor using external-capturing logic.")]
+		public void ExternalCapture(Actor target)
```
It would be nicer if this was also named `Capture`, the same way that `Mobile` and `Aircraft` expose alternative `Move` script properties, and the different versions of `Attack`.

abcdefg30 commented:
Updated.

Mailaender commented on OpenRA.Mods.Common/OpenRA.Mods.Common.csproj:
```diff
@@ -198,6 +198,7 @@
     <Compile Include="Scripting\Global\FacingGlobal.cs" />
     <Compile Include="Scripting\Global\HSLColorGlobal.cs" />
     <Compile Include="Scripting\Global\UserInterfaceGlobal.cs" />
+    <Compile Include="Scripting\Properties\CaptureProperties.cs" />
```
I suggest to move this into a separate pull request as it doesn't provide a test case yet and is independent from infiltrate.

obrakmann commented on OpenRA.Mods.Common/Scripting/Properties/CaptureProperties.cs:
```diff
@@ -0,0 +1,44 @@
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptPropertyGroup("Targeting")]
```
Grouping capturing and infiltration under "Targeting" somehow rubs me the wrong way, but I don't have any better suggestions, either.

Mailaender replied:
We call it special abilities in the tooltips so maybe `Ability`?

obrakmann commented on OpenRA.Mods.RA/Scripting/Properties/InfiltrateProperties.cs:
```diff
@@ -0,0 +1,29 @@
+ */
+#endregion
+
+using System.Linq;
```
I think this isn't needed.

abcdefg30 replied:
Fixed.

obrakmann commented:
Looks good to me otherwise. :+1:

penev92 commented:
I have to agree with the naming nit above about "Targeting".

abcdefg30 committed:
Added an Infiltrate function to lua

abcdefg30 committed:
Make use of the new Infiltrate function in allies05a

abcdefg30 commented:
Updated. Renamed `Targeting` to `Ability`.
> I suggest to move this into a separate pull request as it doesn't provide a test case yet and is independent from infiltrate.

Done: #8614

Mailaender commented:
Works as promised. :white_check_mark:

Mailaender commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/cb06d675a075bb7fe43ed1aa65b7841257550944...1a1d0246e9d44e9d6b86daba4c1412e4ec3ed2f2)
